### PR TITLE
Cpu: Implement Vpadal and Vrintr instructions

### LINKED ISSUE #1781: Unimplemented vrintrne.f64 instruction
### What CPU instruction is missing?

vrintrne.f64

```
Unhandled exception caught: Ryujinx.HLE.Exceptions.UndefinedInstructionException: The instruction at 0x00000000011362c0 (opcode 0x1ef61b60) is undefined!
```

### Instruction name
```
vrintrne.f64 d17, d16
```

### Required by: 

https://github.com/Ryujinx/Ryujinx-Games-List/issues/1027

### PR DESCRIPTION
**Pull request by AcK77**
This PR superseed last instructions left in #2242. Since I'm not a CPU guy I've just ported the code and nothing more. Please be precise during review if there are some changes to be done.

It should fixes #1781

### CODE REVIEW

File: src/ARMeilleure/Instructions/InstEmitSimdCvt32.cs [hunk 1]
```diff
@@ -578,6 +578,12 @@ public static void Vrintp_V(ArmEmitterContext context)
             }
         }
 
+        // VRINTR (floating-point).
+        public static void Vrintr_S(ArmEmitterContext context)
+        {
+            Vrintx_S(context);
```
Comment (gdkchan) [inline src/ARMeilleure/Instructions/InstEmitSimdCvt32.cs:584]: This should be changed, at least to use "FRINTI" on Arm64 fast path. The slow path can do the same as Vrintx since I don't think we emulate inexact exceptions (and the only difference between VRINTX and VRINTR appears to be that the former can raise inexact exceptions while the latter doesn't).

TL;DR: Copy the code from Vrintx_S, change Frintx to Frinti.

File: src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs [hunk 2]
```diff
@@ -511,6 +511,44 @@ private static IEnumerable<ulong> _1D_F_()
...
+        [Test, Pairwise, Description("VRINTR.F<size> <Sd>, <Sm>")]
+        public void Vrintr([Values(0u, 1u)] uint rd,
+                           [Values(0u, 1u)] uint rm,
+                           [Values(2u, 3u)] uint size,
+                           [ValueSource(nameof(_1D_F_))] ulong s0,
+                           [ValueSource(nameof(_1D_F_))] ulong s1,
+                           [ValueSource(nameof(_1D_F_))] ulong s2,
+                           [Values(RMode.Rn, RMode.Rm, RMode.Rp)] RMode rMode)
+        {
+            uint opcode = 0xEB60A40;
```
Comment (gdkchan) [inline src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs:524]:
```suggestion
            uint opcode = 0xEEB60A40;
```
The instruction should have condition set to "Always", otherwise it will not actually execute.

File: src/ARMeilleure/Decoders/OpCodeTable.cs [hunk 3]
```diff
@@ -1018,6 +1019,7 @@ static OpCodeTable()
             SetAsimd("111100111x111010xxxx01101xx0xxxx", InstName.Vrintm,      InstEmit32.Vrintm_V,     OpCode32SimdCmpZ.Create,        OpCode32SimdCmpZ.CreateT32);
             SetAsimd("111100111x111010xxxx01000xx0xxxx", InstName.Vrintn,      InstEmit32.Vrintn_V,     OpCode32SimdCmpZ.Create,        OpCode32SimdCmpZ.CreateT32);
             SetAsimd("111100111x111010xxxx01111xx0xxxx", InstName.Vrintp,      InstEmit32.Vrintp_V,     OpCode32SimdCmpZ.Create,        OpCode32SimdCmpZ.CreateT32);
+            SetAsimd("<<<<11101x110110xxxx101x01x0xxxx", InstName.Vrintr,      InstEmit32.Vrintr_S,     OpCode32SimdS.Create,           OpCode32SimdCmpZ.CreateT32);
```
Comment (gdkchan) [inline src/ARMeilleure/Decoders/OpCodeTable.cs:1022]: This is a "VFP" not an "ASIMD" instruction. It should be in the `// VFP` group and use `SetVfp`.

**Comment by gdkchan** [comment]
The tests are still failing
> UnicornEngine.UnicornEngineException : Invalid instruction (UC_ERR_INSN_INVALID)

Apparently the test is using invalid VRINTR encodings.

File: src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs [hunk 4]
```diff
@@ -511,6 +511,44 @@ private static IEnumerable<ulong> _1D_F_()
...
+                           [Values(RMode.Rn, RMode.Rm, RMode.Rp)] RMode rMode)
+        {
+            uint opcode = 0xEEB60A40;
+
+            V128 v0, v1, v2;
+
+            if (size == 2)
+            {
+                opcode |= ((rm & 0x1e) >> 1) | ((rm & 0x1) << 5);
+                opcode |= ((rd & 0x1e) >> 11) | ((rm & 0x1) << 22);
```
Comment (gdkchan) [inline src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs:531]:
```suggestion
                opcode |= ((rd & 0x1e) << 11) | ((rd & 0x1) << 22);
```

File: src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs [hunk 5]
```diff
@@ -511,6 +511,44 @@ private static IEnumerable<ulong> _1D_F_()
...
+                v0 = MakeVectorE0E1((uint)BitConverter.SingleToInt32Bits(s0), (uint)BitConverter.SingleToInt32Bits(s0));
+                v1 = MakeVectorE0E1((uint)BitConverter.SingleToInt32Bits(s1), (uint)BitConverter.SingleToInt32Bits(s0));
+                v2 = MakeVectorE0E1((uint)BitConverter.SingleToInt32Bits(s2), (uint)BitConverter.SingleToInt32Bits(s1));
+            }
+            else
+            {
+                opcode |= ((rm & 0xf) << 0) | ((rd & 0x10) << 1);
```
Comment (gdkchan) [inline src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs:538]:
```suggestion
                opcode |= ((rm & 0xf) << 0) | ((rm & 0x10) << 1);
```

**Comment by Eli0t77** [comment]
seems to be working fine here so far, played Super Putty Squad for 40mins with no crashes
although there are other graphical issues with the game ill leave them for the compatibility page

[Ryujinx_1.1.0+18d00e8_2024-01-27_15-51-37.log](https://github.com/Ryujinx/Ryujinx/files/14073067/Ryujinx_1.1.0%2B18d00e8_2024-01-27_15-51-37.log)

-  CPU: i7-13700KF
-  GPU: NVIDIA RTX 4070 Ti
-  GPU Driver version: 546.65
-  RAM: 32GB

File: src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs [hunk 6]
```diff
@@ -511,6 +511,45 @@ private static IEnumerable<ulong> _1D_F_()
 
             CompareAgainstUnicorn();
         }
+
+        [Test, Pairwise, Description("VRINTR.F<size> <Sd>, <Sm>")]
+        [Platform(Exclude = "Unix,Linux,MacOsX")]
```
Comment (gdkchan) [inline src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs:516]: What does Unix cover here? If it means "Unix-like", it should cover both Linux and macOS, making them unnecessary?
Also would be nice to get a brief comment of why this test is disabled on Linux and macOS, even if it's just "Causes invalid instruction error on Unicorn emulator.".

Comment (AcK77) [reply src/Ryujinx.Tests/Cpu/CpuTestSimdCvt32.cs:516]: I’ll remove Unix or Linux and macOS then?

**Review by gdkchan — APPROVED** [review]
lgtm, thanks. The Unicorn issue should be probably checked in the future.

**Comment by LDj3SNuD** [comment]
> The Unicorn issue should be probably checked in the future.

[A32] UnicornEngine.UnicornEngineException : Invalid instruction (UC_ERR_INSN_INVALID):
`Vcvt_RM`
`Vrint_RM`
`Vmaxnm_S`
`Vminnm_S`
`Vsel`
`Vmull_I` // P64
`Vrintr_S`
maybe CRC
(My notes, on Windows)

**Review by marysaka — APPROVED** [review]
